indexing-solr/src/test/java/edu/unc/lib/boxc/indexing/solr/filter/SetDatastreamFilterTest.java

```diff
@@ -315,6 +316,62 @@ public void workObjectWithoutPrimaryObjectTest() throws Exception {
         assertNotNull(idb.getFilesizeTotal());
     }
 
+    @Test
+    public void workObjectWithThumbnailNoPrimaryObjectTest() throws Exception {
+        WorkObject workObj = mock(WorkObject.class);
+        when(workObj.getThumbnailObject()).thenReturn(fileObj);
+        when(workObj.getPid()).thenReturn(pid);
+
+        String fileId = "055ed112-f548-479e-ab4b-bf1aad40d470";
+        PID filePid = PIDs.get(fileId);
+        when(fileObj.getPid()).thenReturn(filePid);
+        when(binObj.getPid()).thenReturn(getOriginalFilePid(filePid));
+        setUpDerivatives(filePid);
+
+        dip.setContentObject(workObj);
+        filter.filter(dip);
+
+        assertNotNull(idb.getDatastream());
+        assertNull(idb.getFilesizeSort());
+        assertNotNull(idb.getFilesizeTotal());
+        assertThumbnailDatastreams(fileId);
+    }
+
+    @Test
+    public void workObjectTestWithPrimaryAndThumbnailObjects() throws Exception {
```

Comment:
```suggestion
    public void workObjectWithPrimaryAndThumbnailObjectsTest() throws Exception {
```
This test does raise a good question, what happens if the primary object has thumbnails (which is not set up in this test or the other primary object test at the moment, but is implemented) and there is a thumbnail object set? 

I'd say that we would only want thumbnails from the thumbnail object. So for the time being, maybe when adding thumbnail datastreams we should first delete any existing thumbnail datastreams from the list. Primary objects are semi-defunct at this point, but i don't think we're ready to totally remove them yet.